app/main/views/unsubscribe_requests.py

```diff
@@ -13,16 +13,27 @@ def unsubscribe_request_reports_summary(service_id):
 
 
 @main.route("/services/<uuid:service_id>/unsubscribe-requests/reports/latest")
-@main.route("/services/<uuid:service_id>/unsubscribe-requests/reports/<uuid:batch_id>")
+@main.route("/services/<uuid:service_id>/unsubscribe-requests/reports/<uuid:batch_id>", methods=["GET", "POST"])
 @user_has_permissions("view_activity")
 def unsubscribe_request_report(service_id, batch_id=None):
     if batch_id:
         report = current_service.unsubscribe_request_reports_summary.get_by_batch_id(batch_id)
     else:
         report = current_service.unsubscribe_request_reports_summary.get_unbatched_report()
-    form = ProcessUnsubscribeRequestForm(is_a_batched_report=report.is_a_batched_report)
+    form = ProcessUnsubscribeRequestForm(is_a_batched_report=report.is_a_batched_report, report_status=report.status)
+
+    if form.validate_on_submit():
+        report_has_been_processed = form.data["report_has_been_processed"]
+        data = {"report_has_been_processed": report_has_been_processed}
+        service_api_client.process_unsubscribe_request_report(service_id, batch_id=batch_id, data=data)
+        message = f"""Report for {format_date_numeric(report.earliest_timestamp)} until  \
+                  {format_date_numeric(report.latest_timestamp)} has been marked as \
```

Comment:
Be good to match the same date format as the `<h1>` here
```suggestion
        message = f"""Report for {format_date_normal(report.earliest_timestamp)} until  \
                  {format_date_normal(report.latest_timestamp)} has been marked as \
```